libtiledbsoma/src/soma/soma_context.h

```diff
@@ -67,13 +72,22 @@ class SOMAContext {
         return cfg;
     }
 
+    std::shared_ptr<ThreadPool>& thread_pool() {
+        return thread_pool_;
+    }
```

Comment:
I'm thinking something like if `thread_pool_ == nullptr` then create it. Right now you're lazy loading `SOMAContext` in the reindexer class which does technically lazy load the threadpool. But I don't think you need to lazy load the context in the `IntIndexer`. Just the threadpool in `SOMAContext` if that makes sense.

Reply:
Yes, you are right. that makes sense. I should not load lazily.

Reply:
I would still prefer to see `create_thread_pool` get called in this `SOMAContext` class rather than in the `IntIndexer`. As mentioned above by @bkmartinjr, we'll also want this to be thread-safe - I would probably look into using `std::unique_lock`.

Reply:
I fully agree

Reply:
To clarify, what I'm expecting in the implementation file is something like:
```
    std::shared_ptr<ThreadPool>& SOMAContext::threadpool() {
        std::unique_lock;
        if(thread_pool_ == nullptr)
             create_thread_pool();
        return thread_pool_;
    }
```

Reply:
I agree that we don't actually really need a separate `create_thread_pool()` method though. Was just for the example.

Reply:
I'm on the same page...this is what I was imagining as well.